core/includes/form.inc

```diff
@@ -4973,6 +4990,97 @@ function _form_set_class(&$element, $class = array()) {
   }
 }
 
+/**
+ * Checks the number of input form elements to ensure that the PHP
+ * max_input_vars limit is not exceeded.
+ *
+ * @param array $count
+ *   The number of elements in the form.
+ * @param string $error_level
+ *   The error level for the message displayed on screen if the input vars value
+ *   has been exceeded.
+ *
+ * @see form_pre_render()
+ */
+function _form_check_input_vars($count, $error_level = 'warning') {
+  // Determine the limit on input vars for this server configuration.
+  $limit = ini_get('max_input_vars');
+  if ($limit) {
+    // Leave a 5% buffer for ajax elements or similar.
+    if ($count > $limit * 0.95) {
+      $substitutions = array(
+        '@count' => $count,
+        '@limit' => $limit,
+      );
+
+      // Translatable strings.
+      $warning = t('This form may be too long to work properly.');
+      $notice = t('The last form processed was longer than this system can safely support.');
+      $please_increase = t('This form contains @count input elements. PHP max_input_vars allows only @limit and should be increased.', $substitutions);
+      $data_loss = t('The form has been submitted, but data may heve been lost.');
+
+      switch ($error_level) {
+        case 'warning':
+          if (user_access('administer site configuration')) {
+            $message = $please_increase;
+          }
+          else {
+            $message = $warning;
+          }
+          break;
+
+        case 'error':
+          if (user_access('administer site configuration')) {
+            $message = $data_loss . ' ' . $please_increase;
+          }
+          else {
+            $message = $notice . ' ' . $data_loss;
+          }
+          break;
```

Comment:
@BWPanda the code suggestion above has an empty line with whitespace in it. Can you please fix it, so that it can be merged directly if the change accepted?

Reply:
@klonos Done